lib/mysql_with_async/mysql_with_async.ml

```diff
@@ -0,0 +1,25 @@
+open Async
+
+type dbh = Mysql.dbd
+
+module IO = struct
+  type 'a t = 'a Deferred.t
+
+  let return = Deferred.return
+
+  let bind x f = Deferred.bind x ~f
+end
+
+module Prepared = struct
+  type stmt = Mysql.Prepared.stmt
+
+  type stmt_result = Mysql.Prepared.stmt_result
+
+  let create dbd sql = Deferred.return @@ Mysql.Prepared.create dbd sql
+
+  let close stmt = Deferred.return @@ Mysql.Prepared.close stmt
+
+  let execute_null stmt args = Deferred.return @@ Mysql.Prepared.execute_null stmt args
```

Comment:
Should probably use `thread-pool-async`. Actually all of them, because all are potentially blocking.